### PR TITLE
Headers

### PR DESCRIPTION
**Pull request by ksamborski**
Hi!

I needed some headers support so I added it. Please check if it's fine or not with your idea.

Regards,
Karol

### CODE REVIEW

File: test/test_restclient_get.cpp [hunk 1]
```diff
@@ -49,6 +49,14 @@ TEST_F(RestClientGetTest, TestRestClientGETBodyCode)
   EXPECT_EQ("http://httpbin.org/get", root.get("url", "no url set").asString());
   EXPECT_EQ("restclient-cpp/" VERSION, root["headers"].get("User-Agent", "no url set").asString());
 }
+//check if additional http headers were sent
+TEST_F(RestClientGetTest, TestRestClientGETAdditionalHeaders)
+{
+  RestClient::headermap headers;
+  headers["If-Modified-Since"] = "Sat, 29 Oct 1994 19:43:31 GMT";
+  RestClient::response res = RestClient::get("http://httpbin.org/cache", headers);
+  EXPECT_EQ(304, res.code);
```
Comment (mrtazz) [inline test/test_restclient_get.cpp]: can you change this code to run against the `/headers` endpoint and check the actual return body that the header was set properly? I think that makes for a much better test case and I have the hope that if https://github.com/Runscope/httpbin/issues/248 gets fixed, we can extend the unit tests to the other methods.

**Comment by mrtazz** [comment]
Awesome, thanks so much for this contribution! I left one comment for improving the unit test, if you could fix that, I'll merge it.

**Comment by ksamborski** [comment]
OK, I've fixed that. Please check it.

**Comment by mrtazz** [comment]
@ksamborski awesome, thank you! Can you also merge the current master into your branch? The pull request seems to have a conflict right now.

**Comment by ksamborski** [comment]
I've merged it. I hope it's fine now :)

**Comment by mrtazz** [comment]
awesome! thank you!

**Pull request merged**